### ISSUE TITLE
Internally tagged enum with zero-length struct content does not match

### ISSUE THREAD

**Issue by mqudsi** [serde-rs/json#379]
Given this declaration for `Enum::Foo`:

```rust
#[derive(Serialize, Deserialize, Debug)]
#[serde(tag = "topic")]
enum Enum {
    Foo(Foo)
}

struct Foo;
```

The following json cannot be decoded with `let e: Enum = serde_json::from_str(..)`:

```json
{ "topic": "Foo" }
```

and the following error is returned:

```rust
invalid type: map, expected unit struct Foo
```

Seeing as `Enum::Foo` is internally tagged, shouldn't the above json be a complete representation? More importantly, I think all the needed type info is there such that it can be unambiguously decoded in this case.

**Comment by dtolnay** [comment]
Thanks for the issue! This has been fixed in Serde 1.0.19.

```rust
#[macro_use]
extern crate serde_derive;

extern crate serde;
extern crate serde_json;

#[derive(Serialize, Deserialize, Debug)]
#[serde(tag = "topic")]
enum Enum {
    Foo(Foo)
}

#[derive(Serialize, Deserialize, Debug)]
struct Foo;

fn main() {
    println!("{:?}", serde_json::from_str::<Enum>("{\"topic\":\"Foo\"}").unwrap());
}
```

**Comment by mqudsi** [comment]
Awesome, thanks for fixing it that so fast!
